packages/components/src/EditableItemList.tsx

```diff
@@ -29,25 +29,27 @@ const EditableItemList = (props: EditableItemListProps): React.ReactElement => {
     validate = () => null,
   } = props;
   const [inputError, setInputError] = useState<Error | null>(null);
-  const [selectedIndex, setSelectedIndex] = useState<number | undefined>();
+  const [selectedRanges, setSelectedRanges] = useState<Range[]>([]);
   const [value, setValue] = useState('');
 
-  const handleSelect = useCallback(index => {
-    setSelectedIndex(index);
+  const handleSelectionChange = useCallback((ranges: Range[]) => {
+    setSelectedRanges(ranges);
   }, []);
 
   const handleDelete = useCallback(() => {
-    const newIndex =
-      selectedIndex === items.length - 1 ? items.length - 2 : selectedIndex;
-    if (
-      selectedIndex !== undefined &&
-      selectedIndex >= 0 &&
-      selectedIndex < items.length
-    ) {
-      onDelete(items[selectedIndex]);
-    }
-    setSelectedIndex(newIndex === -1 ? undefined : newIndex);
-  }, [items, selectedIndex, onDelete]);
+    const itemsToDelete: string[] = selectedRanges.reduce(
+      (acc: string[], range: Range) => {
+        const result = [...acc];
+        for (let i = range[0]; i <= range[1]; i += 1) {
+          result.push(items[i]);
+        }
+        return result;
+      },
+      []
+    );
```

Comment:
We could probably put this in `RangeUtils` in the utils package. Seems like this could be common functionality. It's kind of like a flatten.

Reply:
Would `getItemsInRanges` be a good name for the util function?